app/workers/RulesTrigger/function.json

```diff
@@ -2,15 +2,12 @@
   "scriptFile": "__init__.py",
   "bindings": [
     {
-      "authLevel": "function",
+      "authLevel": "anonymous",
```

Comment:
(the rest looks good, really great to have this working)

Reply:
If we put "function" here, I suppose we need the API key in the request, but I have tried adding AZ_RULES_KEY = AZ_FUNCTION_KEY , which is generated by localhost:8000/admin --> tokens, to the header of the PATCH request of ScanReportTableDetailV2/ or to the "code" param of the request, but the request to AZ_URL=http://workers:80 still is not authorised. That's why I have to change it to "anonymous".

I didn't try another key, which I don't have, though. 

Also, I am not sure how this authentication works when Carrot running on the Azure server.

Reply:
Ah I think there is just some undocumented configuration here confusing things.

- `AZ_FUNCTION_KEY` is used by the workers to authenticate with the Django app (so it needs to be generated in the admin backend, I expect we'll be able to remove this soon, as only the upload function depends on it now)
`AZ_RULES_KEY` is used by the Django app to authenticate with the workers (it is generated from within Azure functions UI). When we run the functions locally (not in Docker), I don't think this is used for auth at all as it's a dev mode of sorts maybe? However, it would make sense running them in a container, that auth is necessary. But we can't change the function to allow anonymous, so we'll have a look at a way of doing this by config, or generating a key maybe.

Reply:
So I changed the authLevel to "function", and then we can think more about getting the AZ_RULES_KEY later, maybe in a different PR?